core/aws-core/src/main/java/software/amazon/awssdk/awscore/eventstream/EventStreamAsyncResponseTransformer.java

```diff
@@ -234,15 +239,23 @@ private boolean isException(Message m) {
     }
 
     /**
-     * Transforms an event stream message into a {@link HttpResponse} so we can reuse our existing generated unmarshallers.
+     * Transforms an event stream message into a {@link SdkHttpFullResponse} so we can reuse our existing generated unmarshallers.
      *
-     * @param m Message to transform.
+     * @param message Message to transform.
      */
-    private HttpResponse adaptMessageToResponse(Message m) {
-        HttpResponse response = new HttpResponse(null);
-        response.setContent(new ByteArrayInputStream(m.getPayload()));
-        m.getHeaders().forEach((k, v) -> response.addHeader(k, v.getString()));
-        return response;
+    private SdkHttpFullResponse adaptMessageToResponse(Message message) {
+
+        Map<String, List<String>> headers =
+            message.getHeaders()
+                   .entrySet()
+                   .stream()
+                   .collect(HashMap::new, (m, e) -> m.put(e.getKey(), singletonList(e.getValue().getString())), Map::putAll);
+
+        return SdkHttpFullResponse.builder()
+                                  .content(new AbortableInputStream(new ByteArrayInputStream(message.getPayload()), () -> {
```

Comment:
Can we have a constructor for AbortableInputStream that just takes an input stream? Or perhaps a create method instead.

Reply:
+1 will add